### PR TITLE
tiltfile: don't template tests in helm #1614

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/tiltfile/files.go [hunk 1]
```diff
@@ -339,7 +339,22 @@ func (s *tiltfileState) helm(thread *starlark.Thread, fn *starlark.Builtin, args
 		return nil, fmt.Errorf("Argument 0 (path): %v", err)
 	}
 
-	yaml, err := s.execLocalCmdArgv("helm", "template", localPath.path)
+	templates, err := ioutil.ReadDir(filepath.Join(localPath.path, "templates"))
+	if err != nil {
+		return nil, fmt.Errorf("Error reading chart template directory at %s: %v", localPath.path, err)
```
Comment (nicks) [inline internal/tiltfile/files.go]: i feel like this error message could be a lot better?

Before this change, the error message would come from Helm and look like "no Chart.yaml exists in directory "/home/nick/src/tilt.build/src"", but not you're short-circuiting that

File: internal/tiltfile/files.go [hunk 2]
```diff
@@ -339,7 +339,22 @@ func (s *tiltfileState) helm(thread *starlark.Thread, fn *starlark.Builtin, args
...
+	for _, t := range templates {
+		name := t.Name()
+		if name == "tests" || name == "NOTES.txt" || filepath.Ext(name) == ".tpl" {
+			continue
+		}
+		cmd = append(cmd, "-x")
+		cmd = append(cmd, filepath.Join("templates", name))
+	}
+
+	yaml, err := s.execLocalCmdArgv(cmd...)
```
Comment (nicks) [inline internal/tiltfile/files.go]: should we also adjust the watch to ignore the test files?

File: internal/tiltfile/files.go [hunk 3]
```diff
@@ -339,7 +339,22 @@ func (s *tiltfileState) helm(thread *starlark.Thread, fn *starlark.Builtin, args
...
-	yaml, err := s.execLocalCmdArgv("helm", "template", localPath.path)
+	templates, err := ioutil.ReadDir(filepath.Join(localPath.path, "templates"))
+	if err != nil {
+		return nil, fmt.Errorf("Error reading chart template directory at %s: %v", localPath.path, err)
+	}
+
+	cmd := []string{"helm", "template", localPath.path}
+	for _, t := range templates {
+		name := t.Name()
+		if name == "tests" || name == "NOTES.txt" || filepath.Ext(name) == ".tpl" {
```
Comment (nicks) [inline internal/tiltfile/files.go]: this seems kind of brittle. how confident are you that this is correct?

Comment (nicks) [reply internal/tiltfile/files.go]: poking at the code a bit, i think the actual rule is "ignore NOTES.txt and files that start with underscore"

https://github.com/helm/helm/blob/master/cmd/helm/template.go#L229

i wonder if we should also file a bug report upstream that `helm template` should skip tests by default

Comment (jazzdan) [reply internal/tiltfile/files.go]: Great find! I need to get better at just looking at what open source code does in cases like this.

File: internal/tiltfile/files.go (hunk 2 above)
Comment (jazzdan) [reply internal/tiltfile/files.go]: On second thought, there's no good way to watch a directory with exclusions and we definitely want to watch the directory to catch the case where someone adds a new template to a template directory.

File: internal/tiltfile/tiltfile_test.go [hunk 4]
```diff
@@ -3136,6 +3136,26 @@ update_mode(UPDATE_MODE_MANUAL)
 	f.loadErrString("update_mode can only be called once")
 }
 
+func TestHelmSkipsTests(t *testing.T) {
```
Comment (nicks) [inline internal/tiltfile/tiltfile_test.go]: can you also add a test that asserts an error message if helm() is pointing to a bad directory?

File: internal/tiltfile/files.go (hunk 2 above)
Comment (nicks) [reply internal/tiltfile/files.go]: ok

**Pull request merged**